src/main.ts

```diff
@@ -0,0 +1,115 @@
+/* eslint-disable no-console, @typescript-eslint/no-explicit-any */
+export const LogLevels = {
+  DEBUG: 10,
+  INFO: 20,
+  WARNING: 30,
+  ERROR: 40,
+};
+
+type Tags = Record<string, any>;
+
+export class ProperLogger {
+  name: string;
+
+  commonTags: Tags;
+
+  logLevel: number;
+
+  console: Pick<Console, 'debug' | 'log' | 'warn' | 'error'>;
+
+  static getLogger(name: string): ProperLogger {
+    return new ProperLogger(name);
+  }
+
+  constructor(name: string) {
+    this.name = name;
+    // Here we grab a reference to the `global.console` so that in tests we can
+    // patch this out.
+    this.console = global.console;
+    this.commonTags = {};
+    this.logLevel = LogLevels.INFO;
+    if (process.env.PROPERLY_LOG_LEVEL) {
+      const envValue = parseInt(process.env.PROPERLY_LOG_LEVEL, 10);
+      if (envValue >= 0 && envValue < Number.MAX_SAFE_INTEGER) {
```

Comment:
I think the second check (`< Number.MAX_SAFE_INTEGER`) is redundant, unless the goal was to specifically have that as a "disabled" sentinel value? The return value of `parseInt` will be either `NaN` (which will be rejected by the `envValue >= 0` check) or an integer `<= Number.MAX_SAFE_INTEGER`.

Reply:
I think this is mostly an "is positive integer check", but you're right that the greater than zero should be enough and be less "muddy" (by which I mean currently it seems like the `MAX_SAFE_INTEGER` check is there for a reason).